plugin.audio.rne/resources/lib/lutil.py

```diff
@@ -29,50 +29,59 @@
 
 debug_enable = False # The debug logs are disabled by default.
 
-def set_debug_mode(debug_flag):
-    """This function sets the debug_enable var to log everything if debug option is true."""
-    global debug_enable
-    debug_enable = debug_flag in ("true", True)
 
+def local_log(message):
+    """This function logs the messages into the main KODI log file. Called from the libraries module by other functions."""
 
-def log(message):
-    """This function logs the messages into the main KODI log file. Called from main plugin module."""
     if debug_enable:
-        print "%s" % message
+        print("%s" % message)
```

Comment:
xbmc.log()

Reply:
When the add-on is run under Kodi environment, it always use xbmc.log, and only if the debug flag is activated. This debug function is only invoked under development, out of the Kodi runtime environment. This behaviour is similar to other frameworks, as xbmcswift and others, and eases a lot in the task to debug and find the broken parts whenever the website changes its internal structure. So please, consider to leave this debug function as far as it is never invoked from Kodi.
It is true that the function explanation header leads to confusion as it was originally (Eden) used to log the messages into the Kodi log file, but it is no longer like that. I will modify the function explanation to adapt it to their current purpose.